### PR TITLE
Added support for windows

### PR DESCRIPTION
**Pull request by AlecAivazis**
This PR adds support for both `cmd.exe` and `PowerShell`.

@coryb if you have the time to take a look and make sure I didn't cause any regressions in your recent changes, that would be great! There is still some work to do before I merge this in but I wanted to get this out early since I'm pretty sure I finished the prompts you added.

### CODE REVIEW

File: core/terminal.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+// SoundBell uses the default Bell() behavior in order to not lose functionality.
+func (t *Terminal) SoundBell() {
+	fmt.Fprintf(t, "%c", readline.CharBell)
+	t.Bell()
+}
+
+// DevNull is a writer that writes no where (ie, leave the printing up to me)
+type DevNull struct{}
```
Comment (coryb) [inline core/terminal.go]: probably should be `ioutil.Discard`? https://golang.org/pkg/io/ioutil/#pkg-variables

Comment (AlecAivazis) [inline core/terminal.go]: Definitely - nice catch

**Comment by coryb** [comment]
Hey, I dont fully grok all the changes, but it looks like things are headed in a good direction.  One minor thing that concerns me is making the templates (and template data types) private.  That will prevent users (ie me) from changing the prompts when desired (perhaps I want to change the colors or styling).

The use of `github.com/k0kubun/go-ansi` looks promising, at least a lot of the cursor movement logic looks much cleaner.   I was starting to play with `github.com/gosuri/uilive` which I thought looked promising for eliminating all the cursor logic as well (basically treats the prompt as a print buffer, every time you flush you refresh the prompt).

**Comment by AlecAivazis** [comment]
Ah good point about the templates, I'll make them public. 

Also `go-ansi` has served pretty well in my testing but there are some bugs that I have fixed locally in it's windows support and still need to get them incorporated into the original project. `github.com/gosuri/uilive` looks pretty interesting. Let me know how that goes! 

Once this gets merged in, i'm going to start working on the api we discussed in #31.

**Comment by AlecAivazis** [comment]
@coryb it looks like your name is missing from the Contributors tab on GitHub, not sure if you care but that probably means you're missing some configuration on your local machine to tie your commits to your GH account

**Comment by coryb** [comment]
Thanks for the heads up about the contributors.  Not a big deal, but I was committing with my work git config, it should be remedied the next time I send you a PR :)

**Pull request merged**